spec/lib/vmdb/settings_spec.rb

```diff
@@ -87,6 +87,28 @@
       )
     end
 
+    it "with a change with string keys" do
```

Comment:
2cents:  can this be reworded to something more readable:

```ruby
it "is valid when a change has string keys" do
```

Reply:
FWIW, I personally am not a fan of phrases like "should" "is valid", etc because that's implied by having a test.  Unless I'm writing a test where it's supposed to be invalid or something abnormal.  I think that's also why I don't like constructs like

```
describe "#some_method"
  it "works"
```

Reply:
> Unless I'm writing a test where it's supposed to be invalid or something abnormal.

But the thing is, testing this method does just that above:

```ruby
describe ".save!" do
  let(:miq_server) { FactoryGirl.create(:miq_server) }
 
  it "does not allow invalid configuration values" do
    # ...
```

So you are literally mixing and matching styles here.  Further more, `it` should always be describing output, not input, because I guarantee that all methods you are testing have more than two states (in this case, `valid`/`invalid` or "saved or not saved").  Specifically with this, not sticking to that rule makes it so someone scanning the test names has no idea what it is testing for (as a result, not an input), until they look at the code, or "it's" output is somehow assumed in some unwritten rule that it is "true" when no expected output is written.

* * *

Regarding `it "works"`, I already admitted elsewhere that using vague terminology like that was bad, and fixed my test to reflect that (which you were the reviewer for).  I agree that is vague and should be avoided.  I am fine with the suggestion above where the `it` combines both the `input` and `output` in a single line, versus:

```ruby
describe ".save!" do
  context ""with a change with string keys" do
    it "is valid" do
```

Since it is only testing one thing and the verbosity doesn't add much of anything.  If you are testing more than one output, then I think breaking it up into a context is cleaner and allows you to make use of `let` and `before` constructs to keep tests DRY.